scriptlets/refresh_zapper_if_needed

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# Update the Zapper snap (and firmware) if needed
+#
+# Description:
+#
+# If a Zapper is part of the setup, refresh the snap
+# to the target channel (default is `beta`), upgrade firmware
+# and wait until the add-ons are recognized.
+
+usage() {
+    echo "Usage: $(basename ${BASH_SOURCE[0]}) [--channel]"
+}
+
+CHANNEL=beta
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --channel)
+            CHANNEL=$2
+            shift
+            ;;
+        *)
+            usage
+            echo "Error: Invalid argument $1"
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+# Running in a subshell to not exit with error like it normally would when failing
+(source "$(dirname ${BASH_SOURCE[0]})/defs/check_for_zapper_ip" >> /dev/null)
```

Comment:
I may be missing something but would this work the way you intend it to? If so, I would personally prefer it because it makes the fact that it will exit (if the first command fails) explicit.
```
source "$(dirname ${BASH_SOURCE[0]})/defs/check_for_zapper_ip" >> /dev/null || exit 0
```

Reply:
Unfortunately that still exits with `RC=1` because you're sourcing the file, making it part of the main script. If we make that `check_for_zapper_ip` executable instead, we can make it work like this:
```
$(dirname ${BASH_SOURCE[0]})/defs/check_for_zapper_ip   >> /dev/null || exit 0
```
Let me know if it's ok for you George to make it executable.